client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCache.java

```diff
@@ -155,6 +156,15 @@
     */
    NotifyingFuture<Boolean> replaceWithVersionAsync(K key, V newValue, long version, int lifespanSeconds, int maxIdleSeconds);
 
+   /**
+    * Retrieve entries from the server
+    *
+    * @param filterConverterFactory Factory name for the KeyValueFilterConverter or null
+    * @param segments               The segments to iterate on or null if all segments should be iterated
+    * @param batchSize              The number of entries transferred from the server at a time
+    * @return Iterator for the entries
+    */
+   CloseableIterator<Entry<Object, Object>> retrieveEntries(String filterConverterFactory, Set<Integer> segments, int batchSize);
```

Comment:
@danberindei Good to know. Instead of ```Partitioner```, there's also an alternative name I can use like ```Splitter```.
Wrt to 5465, assuming the HotRod protocol and the hotrod client(s) are patched to work with a custom ```SegmentMapper``` rather than be tied to a ```ConsistentHash```, that won't affect the remote iterator and the subsequent ```Splitter``` as the ConsistentHash is not going to be exposed directly.